File: frontend/public/examples/cpp/primitives.cpp
// Defining primitives in C++

#include <iostream>
#include <string>
#include <vector>
#include <array>

int main() {
    // Integer types
    int age = 25;
    long count = 100L;
    unsigned int positiveOnly = 42;

    // Floating point types
    float temperature = 98.6f;
    double precise = 3.14159265359;

    // Character and string
    char letter = 'A';
    std::string name = "Alice";

    // Boolean
    bool isActive = true;

    // Array
    int numbers[5] = {1, 2, 3, 4, 5};
    std::array<int, 5> stdArray = {1, 2, 3, 4, 5};

    // Vector
    std::vector<int> items = {1, 2, 3};
    items.push_back(4);

    // Pointer
    int* ptr = &age;

    std::cout << "Name: " << name << ", Age: " << age << ", Active: " << isActive << std::endl;
    return 0;
}

File: examples/rust/primitives.rs
// Defining primitives in Rust

fn main() {
    // Integer types
    let age: i32 = 25;
    let count: u64 = 100;

    // Float types
    let temperature: f64 = 98.6;

    // String types
    let name: &str = "Alice";
    let owned_name: String = String::from("Bob");

    // Boolean
    let is_active: bool = true;

    // Character
    let letter: char = 'A';

    // Tuple
    let coordinates: (f64, f64) = (10.0, 20.0);

    // Array
    let numbers: [i32; 5] = [1, 2, 3, 4, 5];

    // Vector
    let mut items: Vec<i32> = vec![1, 2, 3];
    items.push(4);

    println!("Name: {}, Age: {}, Active: {}", name, age, is_active);
}
